### PR TITLE
[SYCL] fix mul_mat fault in CI/unit-test

### PR DESCRIPTION
**Pull request by NeoZhangJianyu**
1. Due to the fault of cpy_f32_f16(), the mul_mat() cases of unit test are fault.
    Fix this issue. 
    Now all CI for SYCL is passed on Arc770.
2. In ci/run.sh, disable rm build folder, to speed up rebuilding CI/unit-test.
3. Support GGML_TYPE_IQ2_XXS, GGML_TYPE_IQ2_XS, GGML_TYPE_IQ3_XXS in mul_mal.

### CODE REVIEW

**Comment by NeoZhangJianyu** [comment]
@airMeng , @luoyu-intel @abhilash1910 could help review this PR?

File: ggml-sycl.cpp [hunk 1]
```diff
@@ -3210,6 +3467,31 @@ typedef struct dpct_type_107281 {
 } block_q6_K;
 static_assert(sizeof(block_q6_K) == sizeof(ggml_fp16_t) + 13*QK_K/16, "wrong q6_K block size/padding");
 
+#define QR2_XXS 8
+#define QI2_XXS (QK_K / (4*QR2_XXS))
+typedef struct dpct_type_101531 {
```
Comment (abhilash1910) [inline ggml-sycl.cpp:3472]: Do we need ".._101531" or it can be only struct dpct_type_<quant process such as QR2_XS>

File: ci/run.sh [hunk 2]
```diff
@@ -98,7 +98,7 @@ function gg_run {
 function gg_run_ctest_debug {
     cd ${SRC}
 
-    rm -rf build-ci-debug && mkdir build-ci-debug && cd build-ci-debug
+    mkdir -p build-ci-debug && cd build-ci-debug
```
Comment (ggerganov) [inline ci/run.sh:101]: Let's keep the `rm -rf` version because I want to start from a clean build folder each time

**Review by ggerganov — CHANGES_REQUESTED** [review]

File: ggml-sycl.cpp (hunk 1 above)
Comment (NeoZhangJianyu) [reply ggml-sycl.cpp:3472]: yes, rename it.

File: ci/run.sh (hunk 2 above)
Comment (NeoZhangJianyu) [reply ci/run.sh:101]: Yes, I will restore it.

File: ggml-sycl.cpp [hunk 3]
```diff
@@ -9455,24 +10776,385 @@ static void convert_mul_mat_vec_f16_sycl(const void *vx, const dfloat *y,
...
+                        mul_mat_vec_q<QK4_0, QI4_0, block_q4_0,
+                                      VDR_Q4_0_Q8_1_MMVQ, vec_dot_q4_0_q8_1>(
+                            vx, vy, dst, ncols, nrows, item_ct1,
+                            iq3xxs_grid_ptr_ct1, ksigns64_ptr_ct1);
+                    });
+        });
+    }
+}
+
+static void mul_mat_vec_q4_1_q8_1_sycl(const void *vx, const void *vy,
```
Comment (abhilash1910) [inline ggml-sycl.cpp:10809]:
```suggestion
static void mul_mat_vec_q4_1_sycl(const void *vx, const void *vy,
```

File: ggml-sycl.cpp [hunk 4]
```diff
@@ -9455,24 +10776,385 @@ static void convert_mul_mat_vec_f16_sycl(const void *vx, const dfloat *y,
     }
 }
 
-template <int qk, int qi, typename block_q_t, int vdr,
-          vec_dot_q_sycl_t vec_dot_q_sycl>
-static void mul_mat_vec_q_sycl_submitter(const void *vx, const void *vy,
+
+static void mul_mat_vec_q4_0_q8_1_sycl(const void *vx, const void *vy,
```
Comment (abhilash1910) [inline ggml-sycl.cpp:10780]: Can we rewrite the function name as ```mul_mat_vec_q4_0_sycl``` to make it consistent with cuda

File: ggml-sycl.cpp [hunk 5]
```diff
@@ -9189,11 +10372,132 @@ static void dequantize_row_q6_K_sycl(const void *vx, dst_t *y, const int k,
...
+                             });
+        });
+    }
+}
+
+template <typename dst_t>
+static void dequantize_row_iq3_xxs_sycl(const void *vx, dst_t *y, const int k,
+                                        dpct::queue_ptr stream) {
+    const int nb = k / QK_K;
+    {
```
Comment (abhilash1910) [inline ggml-sycl.cpp:10453]: Do we need to have a local scope here?

**Review by abhilash1910 — APPROVED** [review]
LGTM !

File: ggml-sycl.cpp (hunk 3 above)
Comment (NeoZhangJianyu) [reply ggml-sycl.cpp:10809]: This function name is referred to original CUDA code when migration.
I will rename it after the SYCL backend is stable.

File: ggml-sycl.cpp (hunk 4 above)
Comment (NeoZhangJianyu) [reply ggml-sycl.cpp:10780]: This function name is referred to original CUDA code when migration.
I will rename it after the SYCL backend

File: ggml-sycl.cpp (hunk 5 above)
Comment (NeoZhangJianyu) [reply ggml-sycl.cpp:10453]: It's passed in my CI test.
No sure, it will be called with real LLM.

**Review by abhilash1910 — APPROVED** [review]

**Review by ggerganov — APPROVED** [review]

**Pull request merged**